### ISSUE TITLE
Template for standup meetings

### ISSUE THREAD

**Issue by robby1066** [robby1066/keepposted-help#23]
**TL;DR:** This feature would add the ability for people to use [Keep Posted](https://www.keepposted.io) to organize an asynchronous alternative to a daily standup meeting by having team members record their updates via their webcam. Updates would then be compiled into a single message and distributed to everyone on the team. 

# What's the problem you're hoping this new feature will solve? 
Team members may use a 'standup' format to communicate their status and any issues that are blocking them from accomplishing tasks that they're focused on. These meetings are recurring, and may take place daily, weekly, or on another interval.

There are many reasons it may be challenging for a distributed team to gather at the same time. Being able to check in asynchronously may be a useful alternative and has been asked for by several people interested in trying Keep Posted.

## What could be good about asynchronous standup meetings?
- Team members in very different time zones can more easily participate
- More flexibility on when a person wants to engage with the message—it doesn’t have to interrupt a flow state they may be in
- Options for playback at high speed or navigating through updates non-linearly
- People who are not able to make a scheduled meeting for any reason are able to participate
- One less meeting on the calendar—when one less meeting is desirable*

## What could be bad about asynchronous standup meetings?
- Unclear how feedback will be given—If someone says they are blocked, how can someone respond to that with support?
- One less meeting on the calendar—when real-time human contact is desirable*

## What needs to be explored?
- How structured do standup meetings tend to be?
- How many people are involved in standup meetings?
- What actual needs do they fulfill?
- What are the actual frustrations other people experience from them?
- How do managers experience them vs. individual contributors?
- Are standup meetings a chore, rejuvenating, or a bit of both for a team?
- How do experiences of standup meetings differ from team to team?
- What beneficial feedback generally gets given in standup meetings?

# Description of feature
**Initial idea - Subject to change**

A 'Standup' message template in Keep Posted would allow multiple people on the same team to answer a structured question (or set of questions) that would then be compiled into a complete message for sharing among the team at a predictable time.

There should be creator-level control over the questions answered by team members. These questions are likely to be structured and the same questions will be asked to everyone participating. It’s also likely—but not guaranteed—that the same questions will be asked at every standup instance for a team. There may be exceptions to this where a specific question needs to be asked, or the stock questions may be changed now and then.

Example standup meeting questions:

1. What did you do yesterday?
2. What will you do today?
3. Anything blocking your progress?

The person organizing the standup message (the message creator) should be able to easily kick off the process of collecting input from the team. Ideally, message creation and prompting team members for responses can be condensed to a single-action green path (one-click kickoff), with options for making adjustments if needed.

When team input is collected, the organizer should be notified so they can build and distribute the message. Alternatively, there could be an automated cutoff time (‘record your standup update before noon’) after which the message is automatically built and distributed.

# Alternatives and workarounds
Currently, a [collaborative message](https://github.com/robby1066/keepposted-help/wiki/Creating-a-message#Collaborative-messages) could be created asking the same questions to each person. It would be tedious to create if the team was more than a few people, especially if these were a recurring activity.

# Additional context
https://geekbot.com/blog/daily-standup-meeting/
https://www.atlassian.com/team-playbook/plays/standups

**Comment by robby1066** [comment]
Some thoughts on the interaction model that would fit here.

# What roles are involved here?
- **Message organizer:** this is the person who initiates the message and causes the relevant people to be notified about recording a clip
  - This maps to the "message creator" role in other Keep Posted messages. This is the person who the message is ultimately tied to.
  - It's not guaranteed or necessary that the organizer actually records anything for this message. They may just be facilitating and curating
- **Message Participant:** this is one of potentially many people who will record a clip for the message
  - In the context of a standup message, this would include all the people on the team who were giving an update
- **Message Viewer:** this is any of the people who will be viewing—and potentially responding to—the message once it is compiled and distributed
  - This likely includes the Message Organizer and the Message Participants
  - It may also include other people, such as team members who couldn't participate for whatever reason, department heads, or execs, depending on the reporting processes in the organization

# What high level interaction flows make sense?
## Version one
1. Organizer creates the message in Keep Posted
  - Specifies questions to answer—which will be saved as defaults for future messages
2. Organizer distributes link in Slack
3. Participant follows the link
  - they log in or create an account if they don't yet have one
  - they end up on the page prompting them to record their status update
4. Participant records their status update
5. Organizer is notified when all clips are complete
  - **Question:** How does the system know who is expected to record an update?
6. Organizer builds the message
7. Organizer distributes link to the new message
  - Share the link in Slack
  - Share the link in Email

**Comment by robby1066** [comment]
Design Doc: Interaction Flows
https://www.figma.com/proto/8bgbR69YzLLZ6pfEnswtEt/Standup-Message-Concepts?node-id=5%3A2&viewport=766%2C630%2C1&scaling=min-zoom

**Comment by robby1066** [comment]
Thinking through this template, it's becoming apparent that this *could* be abstracted into a more general use-case. It's really useful for any situation where you want to hear a response to a specific question from a group of people. 

The "standup" meeting is just one of those. Others include:
- Recurring status meetings that don't fit the standup model (the name may be alienating to people unfamiliar with scrum?)
- Icebreaker check-ins to help a remote team get to know each other better
- Project postmortems. (what's something you learned from ______?)
- Less frequent general status updates (Describe your OKRs for next quarter)
- Structured feedback or input from customers or clients (In less than a minute, describe whats most valuable thing about our product for you)

There are many others.

How should that be modeled? It's probably not ideal to have the UI say "want to do a standup message? Easy! Just go to the Team check-in, icebreaker, customer research, etc... section."

One option would be to have a generic format that can be presented in multiple contexts on the front end, to shield the users from the complexity and think about their specific task rather than wading through too many use cases.

**Comment by robby1066** [comment]
![image](https://user-images.githubusercontent.com/342510/108558683-e159a280-72bf-11eb-94e0-770f03e9e05b.png)

Example of the template UI (in progress, still needs some cleaning up)

**Comment by robby1066** [comment]
Launched on Feb 26. Still need to update [the documentation](https://github.com/robby1066/keepposted-help/wiki/Creating-a-message). I'll close this as soon as that's done.

**Comment by robby1066** [comment]
Added documentation to the [create messages](https://github.com/robby1066/keepposted-help/wiki/Creating-a-message#roundtable-messages) page in the wiki. 

Closing this issue.
